openassessment/assessment/api/peer.py

```diff
@@ -862,6 +862,107 @@ def set_assessment_feedback(feedback_dict):
         raise PeerAssessmentInternalError(msg)
 
 
+def get_data_for_override_score(submission_uuid, student_item, rubric_dict):
+    """
+    Gets the data needed to render the score override section of the student data page.
+
+    Args:
+        submission_uuid (string): UUID of the student's submission
+        student_item (dict): The dictionary representation of a student item.
+        rubric_dict (dict): The dictionary representation of a rubric
+
+    Returns:
+        scores (dict): points_earned, pointed_possible, points_override
+    """
+    scores = {}
+    score_data = sub_api.get_latest_score_for_submission(submission_uuid)
+    if score_data:
+        scores['points_earned'] = score_data.get('points_earned', None)
+        scores['points_possible'] = score_data.get('points_possible', None)
+
+    # If there is no Score, retrieve points_possible from the rubric as there may not be any assessments yet.
+    if not scores.get('points_possible', None):
+        try:
+            rubric = rubric_from_dict(rubric_dict)
+            points_possible = rubric.points_possible
+        except InvalidRubric:
+            points_possible = None
+
+        scores['points_possible'] = points_possible
+    latest_override_score = sub_api.get_score_override(student_item)
+    if latest_override_score:
+        scores['points_override'] = latest_override_score['points_earned']
+
+    return scores
+
+
+def validate_override_data(points_override, points_possible):
+    """
+    Validate the data provided to the peer_score_override function.
+
+    Args:
+        points_override (string): Points provided by the instructor as an override.
+        points_possible (string): Max points for this question.
+
+    Returns:
+        error message if data check failed.
+    """
+    # If points possible is not a float then the student doesn't yet have a score.
+    try:
+        float(points_possible)
+    except ValueError:
+        return 'An error was encountered creating the override score.'
+
+    try:
+        float(points_override)
+    except ValueError:
+        return 'Please check that you have entered a valid score.'
+
+    if float(points_override) > float(points_possible):
+        return 'You have entered a score greater than the maximum possible.'
+
+    if float(points_override) < 0:
+        return 'You have entered a score less than zero.'
+
+
+def score_override(student_item_dict, points_override, points_possible):
+    """
+    Given a student's ID, override the regular score with a score provided by the instructor.
+
+    Args:
+        student_item_dict (dict): The dictionary representation of a student item.
+        points_override (string): The override points given by the instructor
+        points_possible (string): The max allowed points for this submission.
+    """
+    if student_item_dict.get('student_id', None):
+        msg = validate_override_data(points_override, points_possible)
+        if msg:
```

Comment:
So it looks like this logic works by relying on the "success" case falling-through all of the return clauses and implicitly returning `None`.

At the least, I'd like `validate_override_data` to explicitly return a value. Even then, though, this logic feels a bit opaque. What if instead of returning `str` or `None`, we instead return a tuple of `bool` and `str` like:
```python
# ...
if points_override > points_possible:
    return (False, 'Too high!')
elif points_override < 0:
    return (False, 'Too small!')
else:
    return (True, 'Yay!')
```

then we'd check here like:
```python
success, message = validate_override_data(points_override, points_possible)
if not success:
    return {...}
```

Assuming we implement the above, the method name should probably be changed too.